fabric-gametest-api-v1/src/main/java/net/fabricmc/fabric/mixin/gametest/ArgumentTypesMixin.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.gametest;
+
+import com.mojang.brigadier.arguments.ArgumentType;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+import net.minecraft.SharedConstants;
+import net.minecraft.command.argument.ArgumentTypes;
+import net.minecraft.command.argument.TestClassArgumentType;
+import net.minecraft.command.argument.TestFunctionArgumentType;
+import net.minecraft.command.argument.serialize.ArgumentSerializer;
+import net.minecraft.command.argument.serialize.ConstantArgumentSerializer;
+
+@Mixin(ArgumentTypes.class)
+public abstract class ArgumentTypesMixin {
+	@Shadow
+	public static <T extends ArgumentType<?>> void register(String id, Class<T> argClass, ArgumentSerializer<T> serializer) {
+		throw new AssertionError("Nope.");
+	}
+
+	@Inject(method = "register()V", at = @At("RETURN"))
+	private static void register(CallbackInfo ci) {
+		// Registered by vanilla when isDevelopment is enabled.
+		if (!SharedConstants.isDevelopment) {
```

Comment:
Why is there a "!", i.e. the test arguments are registered when development is *false*??

Reply:
This the vanilla constant, and is almost always false, even in a fabric mod dev env. If/when its true vanilla will register these as noted by the comment.

Reply:
I see.
I currently have the problem that I want to use BungeeCord, but these two test arguments are unknown to BungeeCord and therefore it generates an error (`IndexOutOfBoundsException: Index 49 out of bounds for length 49` at https://github.com/SpigotMC/BungeeCord/blob/master/protocol/src/main/java/net/md_5/bungee/protocol/packet/Commands.java#L595).

Reply:
Okay, I think I have found a workaround: If I exclude the gametest, as described here https://issuehint.com/issue/FabricMC/fabric/2315#84183795, the registration of these test arguments seems to be skipped.
I don't know if this has any side effects .